File: api/sequences.py
from flask import Blueprint, request
from api import TV, RECEIVER, LIGHTS

seq_bp = Blueprint("sequences", __name__)

def chromecast():
    TV.power_on()
    TV.set_input("HDMI-1")
    RECEIVER.send_command("MEDIA")

def switch():
    TV.power_on()
    TV.set_input("HDMI-1")
    RECEIVER.send_command("GAME")

def cubert():
    TV.power_on()
    TV.set_input("HDMI-2")
    RECEIVER.send_command("TV")

def airplay():
    TV.power_on()
    TV.set_input("AirPlay")
    RECEIVER.send_command("TV")

def vinyl():
    RECEIVER.send_command("CD")

def all_off():
    TV.power_off()
    RECEIVER.send_command("POWER")

SEQUENCES = {
    "chromecast": chromecast,
    "switch": switch,
    "cubert": cubert,
    "all_off": all_off,
    "airplay": airplay,
    "vinyl": vinyl
}

@seq_bp.route("", methods=["GET", "POST"])
def sequence():
    if request.method == "GET":
        return {"all_sequences": list(SEQUENCES.keys())}

    if "sequence" not in request.json or {}:
        return {"error": "sequence is required"}, 400

    if sequence in SEQUENCES:
        SEQUENCES[sequence]()
        return "", 204
    else:
        return {"error": "sequence not found"}, 404

File: frontend/src/sequences.ts
import * as Icon from "react-bootstrap-icons";

interface Sequence {
  name: string;
  endpoint: string;
  icon?: Icon.Icon;
  variant?: string;
}

export const sequences: Sequence[] = [
  {
    name: "Chromecast",
    endpoint: "chromecast",
    icon: Icon.Tv,
  },
  {
    name: "Cubert",
    endpoint: "cubert",
    icon: Icon.Cpu,
  },
  {
    name: "Switch",
    endpoint: "switch",
    icon: Icon.Controller,
  },
  {
    name: "Vinyl",
    endpoint: "vinyl",
    icon: Icon.Vinyl,
  },
  {
    name: "AirPlay",
    endpoint: "airplay",
    icon: Icon.Cast,
  },
  {
    name: "I'm done",
    endpoint: "all_off",
    icon: Icon.Power,
    variant: "outline-danger",
  },
];
